Code/Mantid/Framework/Kernel/src/ConfigService.cpp

```diff
@@ -1000,6 +999,44 @@ ConfigServiceImpl::getKeys(const std::string &keyName) const {
   return keyVector;
 }
 
+/**
+ * Recursively gets a list of all config options from a given root node.
+ *
+ * @return Vector containing all config options
+ */
+void ConfigServiceImpl::getKeysRecursive(const std::string &root,
+    std::vector<std::string> &allKeys) const {
+  std::vector<std::string> rootKeys = getKeys(root);
+
+  if(rootKeys.size() == 0)
```

Comment:
I just noticed this, and it's a bit of a nitpick, but Coverity will complain. It suggests
```cpp
if (rootKeys.empty())
```
for performance reasons.